src/v/cluster/cluster_uuid.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/cluster_uuid.h"
+
+#include "storage/api.h"
+
+namespace cluster {
+
+const bytes cluster_uuid_key = "cluster_uuid";
+const storage::kvstore::key_space cluster_uuid_key_space
+  = storage::kvstore::key_space::controller;
+
+std::optional<cluster_uuid> read_stored_cluster_uuid(storage::api& storage) {
+    std::optional<iobuf> cluster_uuid_buf = storage.kvs().get(
+      cluster_uuid_key_space, cluster_uuid_key);
+    if (cluster_uuid_buf) {
+        return cluster_uuid{
+          serde::from_iobuf<uuid_t>(std::move(*cluster_uuid_buf))};
+    }
+    return {};
+}
+
+ss::future<>
+write_stored_cluster_uuid(storage::api& storage, const cluster_uuid& value) {
+    return storage.kvs().put(
+      cluster_uuid_key_space, cluster_uuid_key, serde::to_iobuf(value));
+}
```

Comment:
Should those method be part of `bootstrap_backend` ? Currently they have no context and can be executed from any shard while `bootstrap_backend` is living only on shard 0. This may lead to confusion as kvstore values are shard local

Reply:
Cluster UUID is read from kvstore before `bootstrap_backend` is started, otherwise I would have totally agreed with you. I'm thinking about exposing it from `bootstrap_backend` class as a static function though, so cluster UUID kvstore properties are encapsulated in `bootstrap_backend` and there is no need for the `cluster_uuid.cpp`.

However there is a different issue about kvstore being shard local, I'll chat today about it.

Reply:
This method is a part of `bootstrap_backend` now.